baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/PreferSafeLoggingPreconditions.java

```diff
@@ -0,0 +1,96 @@
+                + "checks from com.palantir.logsafe.Preconditions for standardization as functionality is the same.")
+public final class PreferSafeLoggingPreconditions extends BugChecker implements BugChecker.MethodInvocationTreeMatcher {
+
+    private static final long serialVersionUID = 1L;
+
+    private final Matcher<ExpressionTree> compileTimeConstExpressionMatcher =
+            new CompileTimeConstantExpressionMatcher();
+
+    private static final Matcher<ExpressionTree> PRECONDITIONS_MATCHER = MethodMatchers.staticMethod()
+            .onClass("com.google.common.base.Preconditions")
+            .withNameMatching(Pattern.compile("checkArgument|checkState|checkNotNull"));
+    private static final Matcher<ExpressionTree> OBJECTS_MATCHER = MethodMatchers.staticMethod()
+            .onClass("java.util.Objects")
+            .withNameMatching(Pattern.compile("requireNonNull"));
+
+    @Override
+    public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+        List<? extends ExpressionTree> args = tree.getArguments();
+        if (args.size() > 2) {
+            return Description.NO_MATCH;
+        }
+
+        if (args.size() == 2) {
+            ExpressionTree messageArg = args.get(1);
+            if (!messageArg.getKind().equals(Tree.Kind.STRING_LITERAL)
+                    || !compileTimeConstExpressionMatcher.matches(messageArg, state)) {
+                return Description.NO_MATCH;
+            }
+        }
+
+        if (PRECONDITIONS_MATCHER.matches(tree, state)) {
+            Name methodName = ASTHelpers.getSymbol(tree).name;
+            return buildDescription(tree)
+                    .setMessage("This com.google.common.base.Preconditions call can be replaced with an equivalent "
+                            + "one from com.palantir.logsafe.Preconditions for standardization as the "
+                            + "functionality is the same")
+                    .addFix(SuggestedFix.builder()
+                            .replace(tree, String.format(
+                                    "com.palantir.logsafe.Preconditions.%s(%s)", methodName, tree.getArguments()))
+                            .build())
+                    .build();
+        } else if (OBJECTS_MATCHER.matches(tree, state)) {
+            return buildDescription(tree)
+                    .setMessage("This java.util.Objects call can be replaced with an equivalent "
+                            + "one from com.palantir.logsafe.Preconditions for standardization as the "
+                            + "functionality is the same")
+                    .addFix(SuggestedFix.builder()
+                            .replace(tree, String.format(
+                                    "com.palantir.logsafe.Preconditions.checkNotNull(%s)", tree.getArguments()))
+                            .build())
+                    .build();
```

Comment:
Can we combine these? The message is different, but doesn't need to be because we output a suggested change which illustrates the current and proposed states. We can add a function which maps from a given method name to a safe-logging Preconditions method name instead.

Reply:
👍